source/melpomene/src/main.rs

```diff
@@ -153,77 +152,110 @@ fn kernel_entry(opts: MelpomeneOptions) {
 
     k.initialize(initialization_future).unwrap();
 
-    // Creating a dummy task to use the embedded display driver
-    let graphics_init_future = async move {
-        // Delay for one second, just for funsies
-        Delay::new(Duration::from_secs(1)).await;
+    // Create the interactive console task
+    let graphics_console = async move {
+        // Delay for 1.5 seconds, just for funsies
+        Delay::new(Duration::from_millis(1_500)).await;
 
-        EmbDisplay::register(k, 4, 160, 120).await.unwrap();
+        // Take Port 2 from the serial mux. This corresponds to TCP port 10002 when
+        // you are running crowtty
+        let mut mux_hdl = SerialMuxHandle::from_registry(k).await.unwrap();
+        let p2 = mux_hdl.open_port(2, 1024).await.unwrap();
+        drop(mux_hdl);
 
         let mut disp_hdl = EmbDisplayHandle::from_registry(k).await.unwrap();
-        let mut fc_0 = disp_hdl.get_framechunk(0, 80, 45, 160, 120).await.unwrap();
-        drop(disp_hdl);
-
-        let line_style = PrimitiveStyle::with_stroke(Gray8::WHITE, 1);
-        let tline = Line::new(Point::new(148, 2), Point::new(8, 2)).into_styled(line_style);
-        let bline = Line::new(Point::new(148, 24), Point::new(8, 24)).into_styled(line_style);
-        let lline = Line::new(Point::new(7, 24), Point::new(7, 2)).into_styled(line_style);
-        let rline = Line::new(Point::new(149, 24), Point::new(149, 2)).into_styled(line_style);
-        let text_style = MonoTextStyle::new(&FONT_6X9, Gray8::WHITE);
-        let datetime_style = MonoTextStyle::new(&FONT_5X7, Gray8::WHITE);
-        let text1 = Text::new("Welcome to mnemOS!", Point::new(10, 10), text_style);
-        let text2 = Text::new("A tiny operating system", Point::new(10, 20), text_style);
-        let output_settings = OutputSettingsBuilder::new()
-            .theme(BinaryColorTheme::OledBlue)
-            .build();
-
-        let mut sdisp = SimulatorDisplay::<Gray8>::new(Size::new(320, 240));
-        let mut window = Window::new("mnemOS", &output_settings);
+        let char_y = PROFONT_12_POINT.character_size.height;
+        let char_x = PROFONT_12_POINT.character_size.width + PROFONT_12_POINT.character_spacing;
+
+        // Draw titlebar
+        {
+            let mut fc_0 = disp_hdl.get_framechunk(0, 0, 400, char_y).await.unwrap();
+            let text_style = MonoTextStyle::new(&PROFONT_12_POINT, Gray8::WHITE);
+            let text1 = Text::new(
+                "mnemOS",
+                Point::new(0, PROFONT_12_POINT.baseline as i32),
+                text_style,
+            );
+            text1.draw(&mut fc_0).unwrap();
+
+            let title = "forth shell";
+            let text2 = Text::new(
+                title,
+                Point::new(
+                    400 - ((title.len() as u32) * char_x) as i32,
```

Comment:
maybe worth making the `400` that's referenced in a few places a constant so we don't have to update it all over the place if we change the size of the display?

Reply:
That's a good call. Let me add some consts for that.